File: PodrziMe.Services/Database/TakmicarProfil.cs
using System;
using System.Collections.Generic;

namespace PodrziMe.Services.Database;

public partial class TakmicarProfil
{
    public int TakmicarProfilId { get; set; }

    public string Ime { get; set; } = null!;

    public string Prezime { get; set; } = null!;

    public DateOnly DatumRodjenja { get; set; }

    public int KorisnikId { get; set; }

    public virtual ICollection<Kandidat> Kandidats { get; set; } = new List<Kandidat>();

    public virtual Korisnik Korisnik { get; set; } = null!;
}

File: UI/podrzime_mobile/lib/modals/takmicarProfil.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:podrzime_mobile/modals/korisnik.dart';
import 'package:podrzime_mobile/modals/takmicar.dart';

part 'takmicarProfil.g.dart';

@JsonSerializable()
class TakmicarProfil {
  int? takmicarProfilId;
  String? ime;
  String? prezime;
  DateTime? datumRodjenja;
  int? korisnikId;

  // Relations
  Korisnik? korisnik;
  List<Takmicar>? kandidats;

  TakmicarProfil({
    this.takmicarProfilId,
    this.ime,
    this.prezime,
    this.datumRodjenja,
    this.korisnikId,
    this.korisnik,
    this.kandidats,
  });

  factory TakmicarProfil.fromJson(Map<String, dynamic> json) =>
      _$TakmicarProfilFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$TakmicarProfilToJson(this);

  @override
  String toString() {
    return 'Takmicar('
        'ime: $ime, '
        'prezime: $prezime, '
        'korisnikId: $korisnikId'
        ')';
  }
}
